C/include/c4ListenerTypes.h

```diff
@@ -79,6 +79,8 @@ typedef struct C4ListenerConfig {
     C4String directory;                     ///< Directory where newly-PUT databases will be created
     bool allowCreateDBs;                    ///< If true, "PUT /db" is allowed
     bool allowDeleteDBs;                    ///< If true, "DELETE /db" is allowed
+    bool allowCreateCollections;            ///< If true, "PUT /db.scope.coll" is allowed
```

Comment:
There isn't any precedence for this (the Sync Gateway REST API does not allow this) so I'm wondering what the goal of this is.

Reply:
Well, part of what a collections API does is let you create and delete them. It's a straightforward extension of databases. IIRC there are some SG performance tests that use cblite+REST to sync with SG, and if those tests want to use collections they might well need this.

Reply:
This makes me feel quite awkward because there is a growing set of complex functionality that gets tested on every build, and will block progress if it is not correct, but is not a supported part of the Couchbase Lite binary.  If possible I think it should be removed and built elsewhere if the only thing it is used for is to shoehorn it into a testing framework.

Reply:
The SG REST spec doesn't talk about whether the `/db/` endpoint supports collections. I added [a comment asking about that](https://docs.google.com/document/d/1Jlt6AJeu6lRw7QralS1deizC4sN0Ng1n4ISyt41i1hg/edit?disco=AAAAaR9NckY). I think it has to, otherwise there's no way to use GET to find the doc-count or current sequence of a collection.

Reply:
The GET endpoint supports it, but in SG since you don't directly deal with data the way you do with CBL the delete endpoint is limited to database (or rather, the entry in the SG config).  There is no way to delete or add a collection via SG REST (I asked this to confirm)

Reply:
You're right, SG was never able to create or delete buckets. But CBL can of course manage databases, and the REST API supports that, so I think we should allow the REST API to manage collections too.